### PR TITLE
Documentation ag

### PR DESCRIPTION
**Pull request by grandrea**
I have created a readme file that also works as a bit of a documentation for Xi1 and how to set up the search. Please have a look. I plan to do the same for xiFDR. Some of the stuff is from me using it, most of it is from text in the default config file and information scattered here and there. There is also some background to inform the user on some concepts. Let me know if this is any use.

The whole format of the output csv file is not covered though.

I plan to do the same for xiFDR, which is less annotated.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,28 +1,279 @@
...
 
-XiSearch is implemented as a Java Application. Therefore it requires that java is installed. We recommend to use the the latest update of JAVA version 8.
 
+XiSearch is implemented as a Java Application. Therefore it requires that java is installed. We recommend to use the the latest update of JAVA version 8 or above. 
+The latest version as binary can be downloaded from https://rappsilberlab.org/downloads/
 
-The latest version as binary can be downloaded from
-https://rappsilberlab.org/downloads/
+### Background
+xiSEARCH is a search engine for crosslinking mass spectrometry (crosslinking MS). It supports data acquired with ThermoFisher Orbitrap instruments (.raw format) that have been converted to peak. files (.mgf format), for example with [ProteoWizard MsConvert](https://proteowizard.sourceforge.io/). It then searches the peakfiles against a sequence database in .fasta format to identify crosslinked peptide pairs from mass spectra. The search algorithm uses a target-decoy approach outlined in XXX, which enables false-discovery rate (FDR) estimation. The FDR calculation on the xiSEARCH result is performed by xiFDR.
```
Comment (lutzfischer) [inline README.md:8]:
```suggestion
xiSEARCH is a search engine for crosslinking mass spectrometry (crosslinking MS). It supports data acquired with ThermoFisher Orbitrap instruments (.raw format) that have been converted to peak files (.mgf format), for example with [ProteoWizard MsConvert](https://proteowizard.sourceforge.io/). It then searches the peakfiles against a sequence database in .fasta format to identify crosslinked peptide pairs from mass spectra. The search algorithm uses a target-decoy approach outlined in XXX, which enables false-discovery rate (FDR) estimation. The FDR calculation on the xiSEARCH result is performed by xiFDR.
```

Do you want a reference for target-decoy in general or something that states how it actually works in xi?

File: README.md [hunk 2]
```diff
@@ -1,28 +1,279 @@
...
 
-This includes a startup script for the XiSearch gui.
+xiSEARCH is a flexible search engine that allows for extensive configuration of the search options and of the search scoring methods in crosslink identification. Nevertheless, its design suits best data acquired at high resolution in MS1 and MS2- in the Rappsilber lab, we acquire with 120k resolution in ms1 and 60k resolution ms2. Currently, xiSEARCH does not support MS3 approaches.
```
Comment (lutzfischer) [inline README.md:10]:
```suggestion
xiSEARCH is a flexible search engine that allows for extensive configuration of the search options and of the search scoring methods in crosslink identification. Nevertheless, its design suits best data acquired at high resolution in MS1 and MS2 - in the Rappsilber lab, we acquire with 120k resolution in ms1 and 60k resolution ms2. Currently, xiSEARCH does not support MS3 approaches.
```

File: README.md [hunk 3]
```diff
@@ -1,28 +1,279 @@
...
 
+The xiSEARCH algorithm is described in detail in [Mendez, Fischer *et al.* Mol. Sys. Bio. 2019](https://www.embopress.org/doi/full/10.15252/msb.20198994). The xiSEARCH scoring function is made up of several terms accounting for the goodness of fit of the spectra to the peptide pair selected from the database, including fragment mass error, percentage intensity explained, number of fragments, number of crosslinked fragments.
 
-the Gui provides several tabs to configure your search. 
-In the first tab (Run) the result file (result) and optionally a for writing out full annotations of the spectra (peak annotations) can be defined.
+Scoring happens in three stages: 
+
+1. alpha candidates are selected and scored
+2. top n aplpha candidates are taken and all matching beta-candidates will be selected and prescored
```
Comment (lutzfischer) [inline README.md:17]:
```suggestion
2. top n alpha candidates are taken and all matching beta-candidates (according to the precursormass) will be selected and prescored as pairs
```

File: README.md [hunk 4]
```diff
@@ -1,28 +1,279 @@
...
+3. the top X of these are then fully matched and scored
+
+The scoring function is applied to explain each spectrum with a target-target pair, a target-decoy pair, a decoy-decoy pair, a linear target sequence or a linear decoy sequence. Error control by false discovery rate estimation is then performed in a separate step with xiFDR.
```
Comment (lutzfischer) [inline README.md:20]: at this point xisearch does not care about target or decoy - so it could well be that only decoys are considered or only targets.

Comment (lutzfischer) [reply README.md:20]: Actually it does the full scoring on the top-x alpha-beta-candidates + all possible linear matches

File: README.md (hunk 1 above)
Comment (grandrea) [reply README.md:8]: your call- i would say your 2 publications on T-D?

File: README.md [hunk 5]
```diff
@@ -1,28 +1,279 @@
...
 
-the Gui provides several tabs to configure your search. 
-In the first tab (Run) the result file (result) and optionally a for writing out full annotations of the spectra (peak annotations) can be defined.
+Scoring happens in three stages: 
+
+1. alpha candidates are selected and scored
+2. top n alpha candidates are taken and all matching beta-candidates (according to the precursormass) will be selected and prescored as pairs
+3. the top X of these are then fully matched and scored
+
+The scoring function is applied to explain each spectrum with a target-target pair, a target-decoy pair, a decoy-decoy pair, a linear target sequence or a linear decoy sequence. Error control by false discovery rate estimation is then performed in a separate step with xiFDR.
```
Comment (lutzfischer) [inline README.md:20]:
```suggestion
The scoring function is applied to explain each spectrum with without considering if a peptide is target or decoy. In result for a false positive match the chance to to be target-target, target-decoy or a decoy decoy match should be 1:2:1.  Error control by false discovery rate estimation is then performed in a separate step with xiFDR.
```

Comment (lutzfischer) [reply README.md:20]: Should we add something about decoy generation? A.k. 
```
Decoys are by default generated by reversal of the target proteins followed by shifting of the digestible residues by one amino-acid. If a peptide is already present as a target peptide the decoy peptide sequence will be shuffled. If every permutation is already seen as a target peptide, then the decoy peptide will be dropped. In effect the decoy database can underestimate the false positive rate for very short peptides. This is one of the reasons, why in xiFDR by default matches involving peptides shorter then 5 amino-acids are ignored.
```

File: README.md (hunk 1 above)
Comment (lutzfischer) [reply README.md:8]: added my two FDR papers as reference

File: README.md [hunk 6]
```diff
@@ -1,28 +1,279 @@
...
+
+# Setting up a search in the interface
+
+1. Upload files
+
+The "files" tab allows for the upload of the mass spec data in .mgf format and the database in .fasta format. The path to where the results of the search are written also needs to be set. The decoy database is automatically generated from the uploaded .fasta fikes. However, the user has the liberty to define a custom decoy database by designing one of the fasta files as the decoy database instead.
```
Comment (lutzfischer) [inline README.md:41]:
```suggestion
The "files" tab allows for the upload of the mass spec data in .mgf format and the database in .fasta format. The path to where the results of the search are written also needs to be set. The decoy database is automatically generated from the uploaded .fasta files. However, the user has the liberty to define a custom decoy database by marking one of the FASTA files as the decoy database instead.If a FASTA file is marked as decoy - no additional decoys will be auto-generated. For the correct estimate of FDR for self links and heteromeric links proteins in the target and the decoy database  need to  match each other by having the same accession - just prepending a REV_ for the decoy proteins
```

File: README.md [hunk 7]
```diff
@@ -1,28 +1,279 @@
...
+
+
+2. Setting options - basic config.
+
+We will cover the advanced config in the "setting up a search on the command line" section.
+
+##### crosslinker selection
+Normally, all searches are performed with 2 crosslinkers selected: the crosslinker used in the sample (be it BS3, DSS, SDA or other) and "NonCovalent", which allows the search engine to match spectra with a pair of co-eluting and co-fragmenting linear peptides, which is a common source of misinterpretation of crosslinking MS spectra. Thus, the "multiple" crosslinker box should be ticked and then both the crosslinker of interest and "nonCovalent" (near the bottom) should be selected.
```
Comment (lutzfischer) [inline README.md:49]:
```suggestion
Normally, all searches are performed with 2 crosslinkers selected: the crosslinker used in the sample (be it BS3, DSS, SDA or other) and "NonCovalent", which allows the search engine to match spectra with a pair of co-eluting and co-fragmenting linear peptides that are not actually crosslinked. This is a common source of misinterpretation of crosslinking MS spectra. Thus, the "multiple" crosslinker box should be ticked and then both the crosslinker of interest and "nonCovalent" (near the bottom) should be selected.
```

File: README.md [hunk 8]
```diff
@@ -1,28 +1,279 @@
...
+| Tryp-N                   |                                         |
+| No digestion             | for example used for synthetic peptides |
+
+
+
+#### Miscleavages
+The number of miscleavages to consider in the search. Given that crosslinked peptides generate spectra that are similar to spectra of long, miscleaved linear peptides, we suggest setting this number to 3, or even 4 if the database and set of modifications included in the search is small. This allows for alternative explanations of crosslinked spectra with miscleaved peptides.
+
+#### Number of Threads
+Number of threads to be used for the search. The memory usage scales with the number of threads. If the program runs out of memory, consider re-launching xiSEARCH with increased memory via the -Xmx option (see above).
```
Comment (lutzfischer) [inline README.md:105]:
```suggestion
Number of threads to be used for the search. The memory usage scales with the number of threads. If the program runs out of memory, consider re-launching xiSEARCH with increased memory via the -Xmx option (see above) and/or reduce the number of threads.
```

File: README.md (hunk 5 above)
Comment (lutzfischer) [reply README.md:20]: actually it is about 1:2:1 for both peptides random. for single peptide random it is 1:1:0. Guess that gets to complicated to write in a single sentence

File: README.md [hunk 9]
```diff
@@ -1,28 +1,279 @@
...
 
 In the second tab (Peak Lists) the peak lists to be searched can be defined. Supported formats are mgf-files as generated from msconvert or apl-files as produced by MaxQuant. For mgf-files of other sources the config might need to be adapted to define how to read out a run-name and a scan-number from the TITLE= tags in the files.
 
-The third tab defines the FASTA files to be searched. Several files can be defined. One also define one of the fasta files as custom decoy database. If no decoy database is given one is automatically generated based on the target database.
+The third tab defines the FASTA files to be searched. Multiple files can be defined. 
 
 The fourth tab provides the search configuration. Here all parameters are defined as a in form of a text file. A default config is provided as an example that defines a search with BS3 as cross-linker and Carboxyamidomethylation on Cystein as fixed modification and oxidation of methionine as variable modification. Each option contains a short description of what it does and how to change it.
 
 When everything is configured as desired then start-button on the first tab can be pressed to start the search.
```
Comment (lutzfischer) [inline README.md:33]: That whole section is of an old version. and probably should go

Comment (grandrea) [reply README.md:33]: then it goes :)

File: README.md (hunk 1 above)
Comment (grandrea) [reply README.md:8]: why no more link to rappsilber lab?

Comment (grandrea) [reply README.md:8]: ah github confused me

File: README.md [hunk 10]
```diff
@@ -1,28 +1,279 @@
...
+
+#### Losses
+Which fragment losses to consider in the search.
+
+### Custom config
+Here, additional configurations may be set using the text syntax as in the advanced config or in a config file used by the command line version of xiSEARCH (see next section)
+
+### Do FDR setting 
+If the "Do FDR" box is ticked, xiFDR will automatically be run at the end of xiSEARCH. We tend to leave this option off, as we prefer to run xiFDR in a standalone process to have access to more advanced FDR filtering options.
+
```
Comment (lutzfischer) [inline README.md:127]:
```suggestion

### Changing or adding new entries to the config-gui
The  "BasicConfigEntries.conf" contains all the selectable config values. In this file new entries for crosslinker, enzymes, modifications and losses can be freely defined. The file contains sections for crosslinker, modifications, losses, ions,enzymes and custom settings. Each section has a short description on how to add new entries.

```

Comment (lutzfischer) [reply README.md:127]: Not sure if this is the right place to add this - but I think it should appear somewhere

File: README.md (hunk 4 above)
Comment (grandrea) [reply README.md]: then the same should be changed in the default config, which is where this text comes from. We should also include that figure i mentioned here in the readme...

File: README.md [hunk 11]
```diff
@@ -1,28 +1,279 @@
...
+|--------------------------|-----------------------------------------| 
+| Trypsin                  |                                         |
+| Trypsin/P                | trypsin not restricted by proline       |
+| V8                       |                                         |
+| Lys-C                    |                                         |
+| Lys-C/P                  |                                         |
+| trypsin/P + V8           |                                         |
+| proteinaseK              |                                         |
+| proteinaseK & trypsin\P  |                                         |
+| Chymotrypsin             |                                         |
```
Comment (grandrea) [inline README.md:87]: in the GUI, these options are full of typos and inconsistencies. Chymotrypsin is sometimes chemotrypsin, multiple digestions are sometimes with a slash and other times with a +... consider fixing

File: README.md (hunk 4 above)
Comment (lutzfischer) [reply README.md]: Not sure where you found that text - as for the figure as it is a pull request from your fork - ithink you probably need to add teh file: xi_search_strategy

Comment (grandrea) [reply README.md]: right it is not from there... i remember copy pasting that, though it does read similar to the legend of https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6753376/ fig EV6.

File: README.md [hunk 12]
```diff
@@ -1,28 +1,283 @@
...
+3. the top X of these are then fully matched and scored
+
+The scoring function is applied to explain each spectrum with without considering if a peptide is target or decoy. In result for a false positive match the chance to to be target-target, target-decoy or a decoy decoy match should be 1:2:1.  Error control by false discovery rate estimation is then performed in a separate step with xiFDR.
+
+
+
+### The interface
+the interface provides several tabs to configure your search. In the first tab (Run) the result file (result) and optionally a for writing out full annotations of the spectra (peak annotations) can be defined.
 
 In the second tab (Peak Lists) the peak lists to be searched can be defined. Supported formats are mgf-files as generated from msconvert or apl-files as produced by MaxQuant. For mgf-files of other sources the config might need to be adapted to define how to read out a run-name and a scan-number from the TITLE= tags in the files.
```
Comment (lutzfischer) [inline README.md:27]:
```suggestion
The interface provides several tabs. The first two tabs are the main ones for configuring the search. The first one (Files) defines the input and output, i.e. the the peaklist- and fasta-files to be search and where to write the result. 
The second one (Parameters) configures the actual search. The third one (Feedback) provides the log of the current search and provides a means to contact the developers. The fourth tab contains the change log/Version history. 
```

File: README.md (hunk 11 above)
Comment (lutzfischer) [reply README.md:87]: I will go through the BasicConfigEntries.conf file and have a look

File: README.md [hunk 13]
```diff
@@ -1,28 +1,283 @@
...
 
+XiSearch is implemented as a Java Application. Therefore it requires that java is installed. We recommend to use the the latest update of JAVA version 8 or above. 
+The latest version as binary can be downloaded from https://rappsilberlab.org/downloads/
 
-The latest version as binary can be downloaded from
-https://rappsilberlab.org/downloads/
+### Background
+xiSEARCH is a search engine for crosslinking mass spectrometry (crosslinking MS). It supports data acquired with ThermoFisher Orbitrap instruments (.raw format) that have been converted to peak files (.mgf format), for example with [ProteoWizard MsConvert](https://proteowizard.sourceforge.io/). It then searches the peakfiles against a sequence database in .fasta format to identify crosslinked peptide pairs from mass spectra. The search algorithm uses a target-decoy approach outlined in  [Fischer _et al._ 2017](https://doi.org/10.1021/acs.analchem.6b03745) and [Fischer _et al._ 2018]([url](https://doi.org/10.1371%2Fjournal.pone.0196672))  , which enables false-discovery rate (FDR) estimation. The FDR calculation on the xiSEARCH result is performed by xiFDR.
```
Comment (lutzfischer) [inline README.md:8]: I would phrase that less stringent to thermo only. It was also used with e.g. timstof acquisitions.
```suggestion
xiSEARCH is a search engine for crosslinking mass spectrometry (crosslinking MS). It is mainly tested with data acquired with ThermoFisher Orbitrap instruments (.raw format) that have been converted to peak files (.mgf format), for example with [ProteoWizard MsConvert](https://proteowizard.sourceforge.io/) - but any highresolution data in mgf-format are likely to be usable. It then searches the peakfiles against a sequence database in .fasta format to identify crosslinked peptide pairs from mass spectra. The search algorithm uses a target-decoy approach outlined in  [Fischer _et al._ 2017](https://doi.org/10.1021/acs.analchem.6b03745) and [Fischer _et al._ 2018]([url](https://doi.org/10.1371%2Fjournal.pone.0196672))  , which enables false-discovery rate (FDR) estimation. The FDR calculation on the xiSEARCH result is performed by xiFDR.
```

**Pull request merged**